generators/swift/base/src/asIs/Sources/HTTPClient.swift

```diff
@@ -245,38 +253,119 @@ final class HTTPClient: Sendable {
     }
 
     private func executeRequestWithURLSession(
-        _ request: URLRequest
+        _ request: URLRequest,
+        requestOptions: RequestOptions? = nil
     ) async throws -> (Data, String?) {
-        do {
-            let (data, response) = try await clientConfig.urlSession.data(for: request)
+        let maxRetries = requestOptions?.maxRetries ?? clientConfig.maxRetries
+        var lastResponse: (Data, HTTPURLResponse)?
 
-            guard let httpResponse = response as? HTTPURLResponse else {
-                throw ClientError.invalidResponse
-            }
+        for attempt in 0...maxRetries {
+            do {
+                let (data, response) = try await clientConfig.urlSession.data(for: request)
+
+                guard let httpResponse = response as? HTTPURLResponse else {
+                    throw ClientError.invalidResponse
+                }
+
+                // Handle successful responses
+                if 200...299 ~= httpResponse.statusCode {
+                    let contentType = httpResponse.value(forHTTPHeaderField: "Content-Type")
+                    return (data, contentType)
+                }
+
+                lastResponse = (data, httpResponse)
+
+                if attempt < maxRetries && shouldRetry(statusCode: httpResponse.statusCode) {
+                    let delay = getRetryDelay(response: httpResponse, retryAttempt: attempt)
+                    try await Task.sleep(nanoseconds: UInt64(delay * 1_000_000_000))
+                    continue
+                }
+
+                // Handle error responses (no more retries)
+                try handleErrorResponse(
+                    statusCode: httpResponse.statusCode,
+                    data: data
+                )
 
-            // Handle successful responses
-            if 200...299 ~= httpResponse.statusCode {
+                // This should never be reached, but satisfy the compiler
```

Comment:
wow yeah I am surprised this is needed since `handleErrorResponse` is typed as throws.

Is there more way to indicate "this function will never return it throws 100% of the time)